File: Lecture 6/FinalCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/Movie.java
package com.inovaeg.androidtrainingpharos2020;

import com.google.gson.annotations.SerializedName;

public class Movie {
    private @SerializedName("id")
    int id;
    private @SerializedName("title")
    String name;
    private @SerializedName("poster_path")
    String posterUrl;
    private @SerializedName("overview")
    String overview;

    public Movie(String name, String posterUrl, String overview) {
        this.name = name;
        this.posterUrl = posterUrl;
        this.overview = overview;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPosterUrl() {
        return posterUrl;
    }

    public void setPosterUrl(String posterUrl) {
        this.posterUrl = posterUrl;
    }

    public String getOverview() {
        return overview;
    }

    public void getOverview(String overview) {
        this.overview = overview;
    }
}

File: Lecture 6/StarterCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/Movie.kt
package com.inovaeg.androidtrainingpharos2020

import com.google.gson.annotations.SerializedName
import java.util.*

// Class With it's Constructor, Setters & Getters
data class Movie(@SerializedName("original_title") var name: String,
                 @SerializedName("poster_path") var posterUrl: String,
                 @SerializedName("overview") var overview: String,
                 @SerializedName("id") var id: Int = 0) {

    override fun toString(): String {
        return "Movie{" + "name='" + name  + '\'' +
                ", image='" + posterUrl  + '\'' +
                ", overview='" + overview  + '\'' +
                ", id='" + id  + '\'' + "}"
    }

}